libpdb/topic.go

```diff
@@ -1,87 +1,100 @@
 package libpdb
 
 import (
-	"crypto/aes"
-	"crypto/cipher"
+	"bytes"
 	"crypto/rand"
-	"crypto/sha1"
 	"encoding/binary"
+	"encoding/gob"
+	"github.com/agl/ed25519"
 	"github.com/dchest/siphash"
 	"github.com/ryscheng/pdb/bloom"
 	"github.com/ryscheng/pdb/common"
 	"github.com/ryscheng/pdb/drbg"
-	"golang.org/x/crypto/pbkdf2"
+	"golang.org/x/crypto/nacl/box"
 )
 
 type Topic struct {
-	Id      uint64
-	Seed1   drbg.Seed
-	Seed2   drbg.Seed
-	EncrKey []byte
-	// for PIR
-	drbg *drbg.HashDrbg
-	// for use with KDF
-	salt       []byte
-	iterations int
-	keyLen     int
+
+	// For locating log entries
+	Id    uint64
+	Seed1 drbg.Seed
+	Seed2 drbg.Seed
+
+	// For encrypting / decrypting messages
+	sharedSecret *[32]byte
+	// For authenticity
+	// TODO: this should ratchet.
+	signingPrivateKey *[64]byte
```

Comment:
Shouldn't this be outside of the topic handle? Maybe we should move the common parts out and call it TopicHandle or something